### PR TITLE
Support different versions of CRTM and FMS for JEDI and UFS

### LINKED ISSUE #292: JEDI and UFS need different versions of CRTM and FMS
**Is your feature request related to a problem? Please describe.**

The JEDI and UFS applications currently need different versions of CRTM and FMS. While we are working on resolving the differences in the code, for now we need a way to request the correct version for each of the applications.

This has been done in the past for FMS, but the solution was incomplete and sometimes led to building the JEDI FMS tag for the ufs-weather-model.

**Describe the solution you'd like**

The `jedi-{fv3,ufs}-env` and `ufs-weather-model-env` packages need to request the correct versions of FMS and CRTM.

### PR DESCRIPTION
**Pull request by climbfuji**
- Update submodule pointer for spack for changes in https://github.com/NOAA-EMC/spack/pull/144
    -  Need merge of https://github.com/NOAA-EMC/spack/pull/144 and then subsequent submodule pointer update
- Add crtm@v2.3-jedi.4 to configs/templates/skylab-dev/spack.yaml

Fixes https://github.com/NOAA-EMC/spack-stack/issues/292

Tested on my macOS for the various combinations: request both `jedi-ufs-env` and `ufs-weather-model-env`, or just one of them.

### CODE REVIEW

File: .gitmodules
Comment (mer-a-o) [inline .gitmodules]: Do we need these changes?

Comment (climbfuji) [reply .gitmodules]: Temporarily, yes. That's how to test dependent PRs with git submodules. Transparent and easy.

**Review by mer-a-o — APPROVED** [review]

**Comment by climbfuji** [comment]
Waiting for CI tests to pass before merging the spack PR and then this one.

**Review by kgerheiser — APPROVED** [review]

**Pull request merged**
